src/servers/multi_server.cc

```diff
@@ -0,0 +1,947 @@
+        } else if (!strcmp(optarg, "gpu")) {
+          requested_memory_type = TRITONSERVER_MEMORY_GPU;
+        } else {
+          Usage(
+              argv,
+              "-m must be used to specify one of the following types:"
+              " <\"system\"|\"pinned\"|gpu>");
+        }
+        break;
+      }
+      case 'r':
+        model_repository_paths.push_back(optarg);
+        break;
+      case 'v':
+        verbose_level = 1;
+        break;
+      case '?':
+        Usage(argv);
+        break;
+    }
+  }
+
+  if (model_repository_paths.size() != 3) {
+    Usage(
+        argv,
+        "-r must be used to specify three model repository paths, 2 unique and "
+        "1 common");
+  }
+  for (const auto& repo_path : model_repository_paths) {
+    if (repo_path.empty()) {
+      Usage(argv, "model repository path must not be empty");
+    }
+  }
+#ifndef TRITON_ENABLE_GPU
+  if (enforce_memory_type && requested_memory_type != TRITONSERVER_MEMORY_CPU) {
+    Usage(argv, "-m can only be set to \"system\" without enabling GPU");
+  }
+#endif  // TRITON_ENABLE_GPU
+
+  // Check API version.
+  uint32_t api_version_major, api_version_minor;
+  FAIL_IF_ERR(
+      TRITONSERVER_ApiVersion(&api_version_major, &api_version_minor),
+      "getting Triton API version");
+  if ((TRITONSERVER_API_VERSION_MAJOR != api_version_major) ||
+      (TRITONSERVER_API_VERSION_MINOR > api_version_minor)) {
+    FAIL("triton server API version mismatch");
+  }
+
+  // Create two instances of the server with 1 common and 1 unique repo each
+  TRITONSERVER_ServerOptions *server1_options = nullptr,
+                             *server2_options = nullptr;
+  SetServerOptions(
+      &server1_options, verbose_level, model_repository_paths[0],
+      model_repository_paths[1]);
+  SetServerOptions(
+      &server2_options, verbose_level, model_repository_paths[0],
+      model_repository_paths[2]);
+
+  TRITONSERVER_Server *server1_ptr = nullptr, *server2_ptr = nullptr;
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerNew(&server1_ptr, server1_options),
+      "creating first server instance");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerNew(&server2_ptr, server2_options),
+      "creating second server instance");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerOptionsDelete(server1_options),
+      "deleting server options");
+  FAIL_IF_ERR(
+      TRITONSERVER_ServerOptionsDelete(server2_options),
+      "deleting server options");
+
+  std::shared_ptr<TRITONSERVER_Server> server1(
+      server1_ptr, TRITONSERVER_ServerDelete);
+  std::shared_ptr<TRITONSERVER_Server> server2(
+      server2_ptr, TRITONSERVER_ServerDelete);
+
+  // Wait and until the servers are both live and ready.
+  CheckServerLiveAndReady(server1);
+  CheckServerLiveAndReady(server2);
+
+  // Print status of the servers.
+  PrintServerStatus(server1);
+  PrintServerStatus(server2);
+
+  // Load models in both servers.
+  FAIL_IF_ERR(
```

Comment:
`TRITONSERVER_ServerLoadModel` is currently blocking right?
Will it not be a better idea to have the load/unload calls to server1 and server2 concurrently for this testing purpose?
May be we catch any synchronizing issues with model_repository?